docs/diagnostics/overview.md

```diff
@@ -2,97 +2,232 @@
 
-```xml
-<PropertyGroup>
-  <NoWarn>$(NoWarn);ASPIREHOSTINGPYTHON001</NoWarn>
-<PropertyGroup>
+There are some common patterns for suppressing diagnostics in .NET projects. The best method depends on your context and the specific diagnostic. Here's a quick guide to help you choose:
+
+#### The suppress message attribute
+
+The <xref:System.Diagnostics.CodeAnalysis.SuppressMessageAttribute> is ideal when you need targeted, documented suppression tied directly to a specific code element like a class or method. It shines when you’re making a deliberate exception to a rule that is valid in most other places. This attribute keeps the suppression close to the code it affects, which helps reviewers and future maintainers understand the rationale. While it's a clean solution for isolated cases, it can clutter the code if overused, and it’s not the best choice for widespread or bulk suppressions.
+
+#### The project file
+
+Using the `NoWarn` property in the project file is great when you need to suppress a diagnostic across the entire project. This is helpful in cases where a diagnostic is irrelevant to your scenario or when dealing with known false positives. It’s a simple, central way to silence a diagnostic without touching your code. However, it lacks visibility—developers won’t see that a diagnostic is suppressed unless they inspect the project file—so it can make it harder to track intentional exceptions or identify the reason behind suppressions.
+
+#### Preprocessor directives
+
+Preprocessor directives like `#pragma warning disable` offer fine-grained suppression within a specific scope, such as a method body or block of code. They're especially handy when you need a temporary suppression during refactoring, or when a rule incorrectly flags a particular line that you can't or don't want to change. The ability to tightly wrap just the affected code makes this approach powerful, but it can also make the code harder to read and maintain, especially if the directives are scattered or forgotten over time.
+
+#### The editor configuration
+
+Suppressing diagnostics using an _.editorconfig_ file is ideal for enforcing or adjusting analyzer behavior at scale. It allows teams to standardize severity levels (or disable rules) across an entire solution or per directory/file pattern. This method keeps suppression cleanly out of the codebase, and it works well for communicating team-wide conventions. However, it can be a bit opaque—developers need to know to look in the config file—and it doesn't offer the precision of an attribute or pragma when dealing with one-off cases.
+
+### Summary of suppression methods
+
+When deciding how to suppress a diagnostic, consider the scope of the suppression, the visibility you want for the suppression, and the best use case for each method. Here's a summary of the methods available:
+
+| Method                    | Scope                 | Best for                        | Visibility           |
```

Comment:
I'm not sure I understand the "Visibility" column. Why is the attribute higher visibilty than #pragma? Why is the editorconfig higher visibility than in the .csproj (or props/targets files)?